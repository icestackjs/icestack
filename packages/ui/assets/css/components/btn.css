.btn-primary {
  @apply border-primary bg-primary text-primary-content outline-primary;
}
.btn-success {
  @apply bg-success;
}
.btn-warning {
  @apply bg-warning;
}
.btn-error {
  @apply bg-error;
}