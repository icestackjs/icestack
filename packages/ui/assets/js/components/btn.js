module.exports = {
  ".btn-primary": {
    "@apply border-primary bg-primary text-primary-content outline-primary": true
  },
  ".btn-success": {
    "@apply bg-success": true
  },
  ".btn-warning": {
    "@apply bg-warning": true
  },
  ".btn-error": {
    "@apply bg-error": true
  }
}