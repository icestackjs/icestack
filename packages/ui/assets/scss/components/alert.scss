.alert {
  @apply rounded-box text-base-content border-base-200 border p-4;

  &-info {
    @apply text-info-content border-info/20 bg-info;
  }
  &-success {
    @apply text-success-content border-success/20 bg-success;
  }
  &-warning {
    @apply text-warning-content border-warning/20 bg-warning;
  }
  &-error {
    @apply text-error-content border-error/20 bg-error;
  }
}
