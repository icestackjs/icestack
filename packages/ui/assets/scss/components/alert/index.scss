@use "../../utils/index.scss" as *;
@use "sass:map";
$TypeMap: inject("alert.colors");
$DefaultMap: inject("alert.defaults.styled");

.alert {
  @include resolve(map.get($DefaultMap, "default"));

  @each $typeName, $map in $TypeMap {
    &#{getSelector($typeName)} {
      @include resolve(map.get($map, "default"));
    }
  }
}
