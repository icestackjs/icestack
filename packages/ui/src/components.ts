export { componentNames as names, removeDefaultComponents, schemaMap } from '@icestack/preset-default/components'
