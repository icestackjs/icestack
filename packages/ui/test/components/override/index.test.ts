// import { omit } from 'lodash-es'
import { createContext } from '@/context'
import { removeDefaultComponents } from '@/components'
describe('override', () => {
  it('override btn', async () => {
    const components = {
      ...removeDefaultComponents,
      button: {
        override: {
          base: `
          .btn{
            @apply rounded-btn inline-flex flex-shrink-0 cursor-pointer select-none flex-wrap items-center justify-center border-transparent text-center transition duration-200 ease-out;

            &-disabled,
                &[disabled],
                &:disabled{
                  @apply bg-gray-100;
                }
          }
          `
        }
      }
    }
    const ctx = createContext({
      components,
      dryRun: true
    })
    const res = await ctx.buildComponents()
    expect(res.button).toMatchSnapshot()
  })

  it('override btn case 0', async () => {
    const components = {
      ...removeDefaultComponents,
      button: {
        override: {
          base: `.btn{
            @apply rounded-btn inline-flex flex-shrink-0 cursor-pointer select-none flex-wrap items-center justify-center border-transparent text-center transition duration-200 ease-out;

            &-disabled,
                &[disabled],
                &:disabled{
                  @apply bg-gray-100;
                }
          }`
        }
      }
    }
    const ctx = createContext({
      components,
      dryRun: true
    })
    const res = await ctx.buildComponents()
    expect(res.button).toMatchSnapshot()
  })

  it('override btn case 1', async () => {
    const components = {
      ...removeDefaultComponents,
      button: {
        override: {
          base: [
            `.btn{
            @apply rounded-btn inline-flex flex-shrink-0 cursor-pointer select-none flex-wrap items-center justify-center border-transparent text-center transition duration-200 ease-out;

            &-disabled,
                &[disabled],
                &:disabled{
                  @apply bg-gray-100;
                }
          }`
          ]
        }
      }
    }
    const ctx = createContext({
      components,
      dryRun: true
    })
    const res = await ctx.buildComponents()
    expect(res.button).toMatchSnapshot()
  })

  it('override btn case 2', async () => {
    const components = {
      ...removeDefaultComponents,
      button: {
        override: `.btn{
          @apply rounded-btn inline-flex flex-shrink-0 cursor-pointer select-none flex-wrap items-center justify-center border-transparent text-center transition duration-200 ease-out;

          &-disabled,
              &[disabled],
              &:disabled{
                @apply bg-gray-100;
              }
        }`
      }
    }
    const ctx = createContext({
      components,
      dryRun: true
    })
    const res = await ctx.buildComponents()
    expect(res.button).toMatchSnapshot()
  })

  it('override btn case 3', async () => {
    const components = {
      ...removeDefaultComponents,
      button: {
        override: [
          `.btn{
          @apply rounded-btn inline-flex flex-shrink-0 cursor-pointer select-none flex-wrap items-center justify-center border-transparent text-center transition duration-200 ease-out;

          &-disabled,
              &[disabled],
              &:disabled{
                @apply bg-gray-100;
              }
        }`
        ]
      }
    }
    const ctx = createContext({
      components,
      dryRun: true
    })
    const res = await ctx.buildComponents()
    expect(res.button).toMatchSnapshot()
  })

  it('mode btn case 0', async () => {
    const components = {
      ...removeDefaultComponents,
      button: {
        mode: 'none'
      }
    }
    const ctx = createContext({
      mode: 'preset',
      components,
      dryRun: true
    })
    const res = await ctx.buildComponents()
    expect(res.button).toMatchSnapshot()
  })

  it('mode btn case 1', async () => {
    const components = {
      ...removeDefaultComponents,
      button: {
        mode: 'preset'
      }
    }
    const ctx = createContext({
      mode: 'none',
      components,
      dryRun: true
    })
    const res = await ctx.buildComponents()
    expect(res.button).toMatchSnapshot()
  })
})
