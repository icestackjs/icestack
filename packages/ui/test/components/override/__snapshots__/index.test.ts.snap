// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`override > override btn 1`] = `
{
  "base": {
    ".btn": {
      "alignItems": "center",
      "borderColor": "transparent",
      "borderRadius": "var(--ice-rounded-btn, 0.5rem)",
      "cursor": "pointer",
      "display": "inline-flex",
      "flexShrink": 0,
      "flexWrap": "wrap",
      "justifyContent": "center",
      "textAlign": "center",
      "transitionDuration": "200ms",
      "transitionProperty": "color, background-color, border-color, text-decoration-color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter",
      "transitionTimingFunction": [
        "cubic-bezier(0.4, 0, 0.2, 1)",
        "cubic-bezier(0, 0, 0.2, 1)",
      ],
      "userSelect": "none",
    },
    ".btn-disabled,.btn[disabled],.btn:disabled": {
      "--tw-bg-opacity": "1",
      "backgroundColor": "rgb(243 244 246 / var(--tw-bg-opacity))",
    },
  },
  "styled": {
    ".btn": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-base-400) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-base-400) / var(--tw-border-opacity))",
      "borderWidth": "var(--ice-border-btn, 1px)",
      "color": "rgba(var(--ice-base-content) / var(--tw-text-opacity))",
      "outlineColor": "rgba(var(--ice-base-400) / 1)",
      "textDecorationLine": "none",
    },
    ".btn-error": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-error) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-error) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-error-content) / var(--tw-text-opacity))",
      "outlineColor": "rgba(var(--ice-error) / 1)",
    },
    ".btn-error.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-error-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-error-active) / var(--tw-border-opacity))",
    },
    ".btn-error.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-error-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-error-hover) / var(--tw-border-opacity))",
    },
    ".btn-error:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-error-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-error-active) / var(--tw-border-opacity))",
    },
    ".btn-error:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-error-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-error-hover) / var(--tw-border-opacity))",
    },
    ".btn-ghost": {
      "--tw-shadow": "0 0 #0000",
      "--tw-shadow-colored": "0 0 #0000",
      "backgroundColor": "transparent",
      "borderColor": "transparent",
      "borderWidth": "1px",
      "boxShadow": "var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)",
      "color": "currentColor",
      "outlineColor": "currentColor",
    },
    ".btn-ghost.btn-active": {
      "--tw-bg-opacity": "0.2",
      "--tw-border-opacity": "0",
      "backgroundColor": "rgba(var(--ice-base-content) / var(--tw-bg-opacity))",
    },
    ".btn-ghost:active": {
      "--tw-bg-opacity": "0.2",
      "--tw-border-opacity": "0",
      "backgroundColor": "rgba(var(--ice-base-content) / var(--tw-bg-opacity))",
    },
    ".btn-link": {
      "--tw-shadow": "0 0 #0000",
      "--tw-shadow-colored": "0 0 #0000",
      "--tw-text-opacity": "1",
      "backgroundColor": "transparent",
      "borderColor": "transparent",
      "boxShadow": "var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)",
      "color": "rgba(var(--ice-primary) / var(--tw-text-opacity))",
      "outlineColor": "currentColor",
      "textDecorationLine": "underline",
    },
    ".btn-link.btn-active": {
      "backgroundColor": "transparent",
      "borderColor": "transparent",
      "textDecorationLine": "underline",
    },
    ".btn-link:active": {
      "backgroundColor": "transparent",
      "borderColor": "transparent",
      "textDecorationLine": "underline",
    },
    ".btn-neutral": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-neutral) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-neutral) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-neutral-content) / var(--tw-text-opacity))",
      "outlineColor": "rgba(var(--ice-neutral) / 1)",
    },
    ".btn-neutral.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-neutral-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-neutral-active) / var(--tw-border-opacity))",
    },
    ".btn-neutral.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-neutral-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-neutral-hover) / var(--tw-border-opacity))",
    },
    ".btn-neutral:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-neutral-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-neutral-active) / var(--tw-border-opacity))",
    },
    ".btn-neutral:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-neutral-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-neutral-hover) / var(--tw-border-opacity))",
    },
    ".btn-outline": {
      "--tw-shadow": "0 0 #0000",
      "--tw-shadow-colored": "0 0 #0000",
      "--tw-text-opacity": "1",
      "backgroundColor": "transparent",
      "borderColor": "currentColor",
      "boxShadow": "var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)",
      "color": "rgba(var(--ice-base-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-base-1100) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-base-1100) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-base-100) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-error": {
      "--tw-text-opacity": "1",
      "color": "rgba(var(--ice-error) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-error.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-error-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-error-active) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-error-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-error.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-error-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-error-hover) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-error-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-error:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-error-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-error-active) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-error-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-error:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-error-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-error-hover) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-error-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-base-900) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-base-900) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-base-100) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-neutral": {
      "--tw-text-opacity": "1",
      "color": "rgba(var(--ice-neutral) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-neutral.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-neutral-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-neutral-active) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-neutral-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-neutral.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-neutral-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-neutral-hover) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-neutral-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-neutral:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-neutral-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-neutral-active) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-neutral-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-neutral:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-neutral-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-neutral-hover) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-neutral-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-primary": {
      "--tw-text-opacity": "1",
      "color": "rgba(var(--ice-primary) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-primary.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-primary-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-primary-active) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-primary-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-primary.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-primary-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-primary-hover) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-primary-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-primary:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-primary-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-primary-active) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-primary-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-primary:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-primary-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-primary-hover) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-primary-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-success": {
      "--tw-text-opacity": "1",
      "color": "rgba(var(--ice-success) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-success.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-success-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-success-active) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-success-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-success.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-success-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-success-hover) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-success-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-success:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-success-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-success-active) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-success-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-success:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-success-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-success-hover) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-success-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-warning": {
      "--tw-text-opacity": "1",
      "color": "rgba(var(--ice-warning) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-warning.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-warning-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-warning-active) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-warning-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-warning.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-warning-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-warning-hover) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-warning-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-warning:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-warning-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-warning-active) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-warning-content) / var(--tw-text-opacity))",
    },
    ".btn-outline.btn-warning:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-warning-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-warning-hover) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-warning-content) / var(--tw-text-opacity))",
    },
    ".btn-outline:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-base-1100) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-base-1100) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-base-100) / var(--tw-text-opacity))",
    },
    ".btn-outline:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-base-900) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-base-900) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-base-100) / var(--tw-text-opacity))",
    },
    ".btn-primary": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-primary) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-primary) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-primary-content) / var(--tw-text-opacity))",
      "outlineColor": "rgba(var(--ice-primary) / 1)",
    },
    ".btn-primary.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-primary-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-primary-active) / var(--tw-border-opacity))",
    },
    ".btn-primary.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-primary-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-primary-hover) / var(--tw-border-opacity))",
    },
    ".btn-primary:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-primary-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-primary-active) / var(--tw-border-opacity))",
    },
    ".btn-primary:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-primary-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-primary-hover) / var(--tw-border-opacity))",
    },
    ".btn-success": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-success) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-success) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-success-content) / var(--tw-text-opacity))",
      "outlineColor": "rgba(var(--ice-success) / 1)",
    },
    ".btn-success.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-success-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-success-active) / var(--tw-border-opacity))",
    },
    ".btn-success.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-success-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-success-hover) / var(--tw-border-opacity))",
    },
    ".btn-success:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-success-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-success-active) / var(--tw-border-opacity))",
    },
    ".btn-success:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-success-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-success-hover) / var(--tw-border-opacity))",
    },
    ".btn-warning": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "--tw-text-opacity": "1",
      "backgroundColor": "rgba(var(--ice-warning) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-warning) / var(--tw-border-opacity))",
      "color": "rgba(var(--ice-warning-content) / var(--tw-text-opacity))",
      "outlineColor": "rgba(var(--ice-warning) / 1)",
    },
    ".btn-warning.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-warning-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-warning-active) / var(--tw-border-opacity))",
    },
    ".btn-warning.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-warning-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-warning-hover) / var(--tw-border-opacity))",
    },
    ".btn-warning:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-warning-active) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-warning-active) / var(--tw-border-opacity))",
    },
    ".btn-warning:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-warning-hover) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-warning-hover) / var(--tw-border-opacity))",
    },
    ".btn.btn-active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-base-500) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-base-500) / var(--tw-border-opacity))",
    },
    ".btn.btn-disabled,.btn[disabled],.btn:disabled": {
      "--tw-bg-opacity": "0.2",
      "--tw-border-opacity": "0",
      "--tw-text-opacity": "0.2",
      "backgroundColor": "rgba(var(--ice-neutral) / var(--tw-bg-opacity))",
      "color": "rgba(var(--ice-base-content) / var(--tw-text-opacity))",
    },
    ".btn.btn-focus-visible": {
      "outlineOffset": "2px",
      "outlineStyle": "solid",
      "outlineWidth": "2px",
    },
    ".btn.btn-hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-base-300) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-base-300) / var(--tw-border-opacity))",
    },
    ".btn.glass": {
      "--tw-shadow": "0 0 #0000",
      "--tw-shadow-colored": "0 0 #0000",
      "boxShadow": "var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)",
      "outlineColor": "currentColor",
    },
    ".btn.glass.btn-active": {
      "--ice-glass-border-opacity": "15%",
      "--ice-glass-opacity": "25%",
    },
    ".btn.glass:active": {
      "--ice-glass-border-opacity": "15%",
      "--ice-glass-opacity": "25%",
    },
    ".btn:active": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-base-500) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-base-500) / var(--tw-border-opacity))",
    },
    ".btn:focus-visible": {
      "outlineOffset": "2px",
      "outlineStyle": "solid",
      "outlineWidth": "2px",
    },
    ".btn:hover": {
      "--tw-bg-opacity": "1",
      "--tw-border-opacity": "1",
      "backgroundColor": "rgba(var(--ice-base-300) / var(--tw-bg-opacity))",
      "borderColor": "rgba(var(--ice-base-300) / var(--tw-border-opacity))",
    },
  },
  "utils": {
    ".btn-block": {
      "width": "100%",
    },
    ".btn-circle": {
      "borderRadius": "9999px",
      "height": "3rem",
      "padding": "0px",
      "width": "3rem",
    },
    ".btn-circle:where(.btn-lg)": {
      "borderRadius": "9999px",
      "height": "4rem",
      "padding": "0px",
      "width": "4rem",
    },
    ".btn-circle:where(.btn-md)": {
      "borderRadius": "9999px",
      "height": "3rem",
      "padding": "0px",
      "width": "3rem",
    },
    ".btn-circle:where(.btn-sm)": {
      "borderRadius": "9999px",
      "height": "2rem",
      "padding": "0px",
      "width": "2rem",
    },
    ".btn-circle:where(.btn-xs)": {
      "borderRadius": "9999px",
      "height": "1.5rem",
      "padding": "0px",
      "width": "1.5rem",
    },
    ".btn-lg": {
      "fontSize": "1.125rem",
      "height": "4rem",
      "minHeight": "4rem",
      "paddingLeft": "1.5rem",
      "paddingRight": "1.5rem",
    },
    ".btn-md": {
      "fontSize": "0.875rem",
      "height": "3rem",
      "minHeight": "3rem",
      "paddingLeft": "1rem",
      "paddingRight": "1rem",
    },
    ".btn-sm": {
      "fontSize": "0.875rem",
      "height": "2rem",
      "minHeight": "2rem",
      "paddingLeft": "0.75rem",
      "paddingRight": "0.75rem",
    },
    ".btn-square": {
      "height": "3rem",
      "padding": "0px",
      "width": "3rem",
    },
    ".btn-square:where(.btn-lg)": {
      "height": "4rem",
      "padding": "0px",
      "width": "4rem",
    },
    ".btn-square:where(.btn-md)": {
      "height": "3rem",
      "padding": "0px",
      "width": "3rem",
    },
    ".btn-square:where(.btn-sm)": {
      "height": "2rem",
      "padding": "0px",
      "width": "2rem",
    },
    ".btn-square:where(.btn-xs)": {
      "height": "1.5rem",
      "padding": "0px",
      "width": "1.5rem",
    },
    ".btn-wide": {
      "width": "16rem",
    },
    ".btn-xs": {
      "fontSize": "0.75rem",
      "height": "1.5rem",
      "minHeight": "1.5rem",
      "paddingLeft": "0.5rem",
      "paddingRight": "0.5rem",
    },
  },
}
`;
